client/src/main/java/org/apache/cassandra/sidecar/client/SidecarClient.java

```diff
@@ -68,6 +70,32 @@ public SidecarClient(SidecarInstancesProvider instancesProvider,
         executor = requestExecutor;
     }
 
+    /**
+     * Executes the Sidecar health request using the configured selection policy and with no retries
+     *
+     * @return a completable future of the Sidecar health response
+     */
+    public CompletableFuture<HealthResponse> sidecarHealth()
+    {
+        return executor.executeRequestAsync(requestBuilder()
+                .sidecarHealthRequest()
+                .retryPolicy(new NoRetryPolicy())
```

Comment:
this is not quite what we want. The `NoRetryPolicy` will only check a single instance and skip the rest. Ideally, we want to check if any instance is available. I think a `BasicRetryPolicy` with some sensible defaults makes sense.

Reply:
Replaced with `new BasicRetryPolicy(1)`, but it looks to me like the behavior is exactly same…
Also, added unit tests for newly added methods.

Reply:
Should I use the `ExponentialBackoffRetryPolicy` instead?

Reply:
Added a new `OncePerInstanceRetryPolicy` as per Francisco's suggestion.